mlflow/pipelines/steps/split.py

```diff
@@ -244,6 +244,11 @@ def _run(self, output_directory):
             train_df = train_df[post_split_filter(train_df)]
             validation_df = validation_df[post_split_filter(validation_df)]
             test_df = test_df[post_split_filter(test_df)]
+
+        if min(len(train_df), len(validation_df), len(test_df)) < 4:
+            raise MlflowException(
+                "Train, validation, and testing datasets cannot be less than 4 rows."
```

Comment:
Can we output the row length for each dataset and also make the error code `BAD_REQUEST`?